src/dex/miro-migrator/miro-migrator.ts

```diff
@@ -0,0 +1,228 @@
+      return [`${this.dexKey}_${srcToken.address}_${destToken.address}`];
+    }
+
+    return [];
+  }
+
+  async getPricesVolume(
+    srcToken: Token,
+    destToken: Token,
+    amounts: bigint[],
+    side: SwapSide,
+    blockNumber: number,
+    limitPools?: string[],
+  ): Promise<null | ExchangePrices<MiroMigratorData>> {
+    if (!this.isAppropriatePair(srcToken, destToken)) {
+      return null;
+    }
+
+    const state = await this.eventPool.getOrGenerateState(blockNumber);
+    if (!state) return null;
+
+    const prices: bigint[] = [];
+    for (const amount of amounts) {
+      if (amount > state.balance) {
+        prices.push(0n);
+      } else {
+        prices.push(amount);
+      }
+    }
+
+    return [
+      {
+        prices,
+        unit: this.unitPrice,
+        gasCost: MIRO_MIGRATION_GAS_COST,
+        exchange: this.dexKey,
+        poolAddresses: [this.migratorAddress],
+        data: null,
+      },
+    ];
+  }
+
+  getCalldataGasCost(
+    poolPrices: PoolPrices<MiroMigratorData>,
+  ): number | number[] {
+    return CALLDATA_GAS_COST.DEX_NO_PAYLOAD;
+  }
+
+  getAdapterParam(
+    srcToken: string,
+    destToken: string,
+    srcAmount: string,
+    destAmount: string,
+    data: MiroMigratorData,
+    side: SwapSide,
+  ): AdapterExchangeParam {
+    return {
+      targetExchange: this.migratorAddress,
+      payload: '0x',
+      networkFee: '0',
+    };
+  }
+
+  async getSimpleParam(
+    srcToken: string,
+    destToken: string,
+    srcAmount: string,
+    destAmount: string,
+    data: MiroMigratorData,
+    side: SwapSide,
+  ): Promise<SimpleExchangeParam> {
+    const swapData = this.migratorInterface.encodeFunctionData(
+      MiroMigratorFunctions.migratePSPtoXYZ,
+      [srcAmount, '0x'],
+    );
+
+    return this.buildSimpleParamWithoutWETHConversion(
+      srcToken,
+      srcAmount,
+      destToken,
+      destAmount,
+      swapData,
+      this.migratorAddress,
+    );
+  }
+
+  getDexParam(
+    srcToken: Address,
+    destToken: Address,
+    srcAmount: NumberAsString,
+    destAmount: NumberAsString,
+    recipient: Address,
+    data: MiroMigratorData,
+    side: SwapSide,
+  ): DexExchangeParam {
+    const swapData = this.migratorInterface.encodeFunctionData(
+      MiroMigratorFunctions.migratePSPtoXYZ,
+      [srcAmount, '0x'],
+    );
+
+    return {
+      needWrapNative: this.needWrapNative,
+      dexFuncHasRecipient: false,
+      exchangeData: swapData,
+      targetExchange: this.migratorAddress,
+      returnAmountPos: undefined,
+    };
+  }
+
+  async getTopPoolsForToken(
+    tokenAddress: Address,
+    limit: number,
+  ): Promise<PoolLiquidity[]> {
+    if (!this.isPSP(tokenAddress) && !this.isXYZ(tokenAddress)) {
+      return [];
+    }
+
+    return [
+      {
+        exchange: this.dexKey,
+        address: this.migratorAddress,
+        connectorTokens: [
+          {
+            address: this.isPSP(tokenAddress)
```

Comment:
here I am not sure if we can return PSP and a connector for VLR since there's no VRL -> PSP route
@KanievskyiDanylo could you clarify here please